src/main/java/net/dv8tion/jda/internal/requests/restaction/pagination/EntitlementPaginationActionImpl.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.internal.requests.restaction.pagination;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Entitlement;
+import net.dv8tion.jda.api.entities.UserSnowflake;
+import net.dv8tion.jda.api.exceptions.ParsingException;
+import net.dv8tion.jda.api.requests.Request;
+import net.dv8tion.jda.api.requests.Response;
+import net.dv8tion.jda.api.requests.Route;
+import net.dv8tion.jda.api.requests.restaction.pagination.EntitlementPaginationAction;
+import net.dv8tion.jda.api.requests.restaction.pagination.PaginationAction;
+import net.dv8tion.jda.api.utils.data.DataArray;
+import net.dv8tion.jda.api.utils.data.DataObject;
+import net.dv8tion.jda.internal.entities.EntityBuilder;
+import net.dv8tion.jda.internal.utils.Checks;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class EntitlementPaginationActionImpl
+    extends PaginationActionImpl<Entitlement, EntitlementPaginationAction>
+    implements EntitlementPaginationAction
+{
+    protected List<String> skuIds;
+    protected long guildId;
+    protected long userId;
+    protected boolean excludeEnded;
+
+    public EntitlementPaginationActionImpl(JDA api)
+    {
+        super(api, Route.Applications.GET_ENTITLEMENTS.compile(api.getSelfUser().getApplicationId()), 1, 100, 100);
+        this.skuIds = new ArrayList<>();
+        this.guildId = 0;
+        this.userId = 0;
+    }
+
+    @Nonnull
+    @Override
+    public EntitlementPaginationAction order(@Nonnull PaginationAction.PaginationOrder order)
```

Comment:
This can be replaced with getSupportedOrders:

```java
    @Nonnull
    @Override
    public EnumSet<PaginationOrder> getSupportedOrders()
    {
        return EnumSet.of(PaginationOrder.BACKWARD, PaginationOrder.FORWARD);
    }
```

Reply:
I assume I'm only specifying BACKWARD as the supported order due to the issue I mentioned to you in DMs about FORWARD causing infinite requesting

Reply:
@MinnDevelopment can you confirm? So we can finish this off

Reply:
I was able to make it work with my change, removing the order override is necessary.